src/Featurizers/TruncatedSVDFeaturizer.h

```diff
@@ -0,0 +1,631 @@
+        using SVDComponentsEstimator                     = SVDComponentsEstimator<InputT, TransformedT, MaxNumTrainingItemsV>;
+        // ----------------------------------------------------------------------
+
+        SVDComponentsAnnotationData const &              data(SVDComponentsEstimator::get_annotation_data(this->get_column_annotations(), _colIndex, SVDComponentsEstimatorName));
+
+        return typename BaseType::TransformerUniquePtr(new SVDTransformer<InputT, TransformedT>(data.SingularValues, data.SingularVectors));
+    }
+};
+
+} // namespace Details
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         SVDFeaturizer
+///  \brief         Creates the `SVDTransformer` object.
+///
+template <
+    typename InputT,
+    typename TransformedT,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class SVDEstimator :
+    public Components::PipelineExecutionEstimatorImpl<
+        SVDComponentsEstimator<InputT, TransformedT, MaxNumTrainingItemsV>,
+        Details::SVDEstimatorImpl<InputT, TransformedT, MaxNumTrainingItemsV>
+    > {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using BaseType =
+        Components::PipelineExecutionEstimatorImpl<
+            SVDComponentsEstimator<InputT, TransformedT, MaxNumTrainingItemsV>,
+            Details::SVDEstimatorImpl<InputT, TransformedT, MaxNumTrainingItemsV>
+        >;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    SVDEstimator(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex);
+    ~SVDEstimator(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(SVDEstimator);
+};
+
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// |
+// |  Implementation
+// |
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+// ----------------------------------------------------------------------
+
+// ----------------------------------------------------------------------
+// |
+// |  SVDComponentsAnnotationData
+// |
+// ----------------------------------------------------------------------
+template <typename T>
+SVDComponentsAnnotationData<T>::SVDComponentsAnnotationData(T singularvalues, T singularvectors) :
+    SingularValues(
+        std::move(
+            [&singularvalues](void) -> T & {
+                if(singularvalues.size() == 0)
```

Comment:
singularvalues.empty() is probably a more clear description of the check being done here.

Reply:
Eigen::Matrix does not have a empty() method